app/src/main/java/ru/vssemikoz/newsfeed/MainActivity.java

```diff
@@ -5,34 +5,65 @@
 import androidx.recyclerview.widget.RecyclerView;
 
 import android.os.Bundle;
+import android.util.Log;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
 import ru.vssemikoz.newsfeed.adapters.NewsFeedAdapter;
+import ru.vssemikoz.newsfeed.api.NewsApi;
 import ru.vssemikoz.newsfeed.models.NewsItem;
+import ru.vssemikoz.newsfeed.models.NewsItemList;
 
 public class MainActivity extends AppCompatActivity {
-
+    String KEY = "c94a57cbbb50497f94a2bb167dc91fc5";
     List<NewsItem> newsItems = new ArrayList<NewsItem>();
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        initData();
+        Callback<NewsItemList> callbackNewsItemList = new Callback<NewsItemList>() {
+            @Override
+            public void onResponse(Call<NewsItemList> call, Response<NewsItemList> response) {
+                if (!response.isSuccessful()){
+                    Log.d("MyLog", "onResponse " + response.code());
+                    return;
+                }
+                newsItems = Objects.requireNonNull(response.body()).getNewsItem();
+                Log.d("MyLog", "onSuccess " + newsItems);
+                initRecView();
+            }
+
+            @Override
+            public void onFailure(Call<NewsItemList> call, Throwable t) {
+                Log.d("MyLog", "onFailure " + Objects.requireNonNull(t.getMessage()));
+
+            }
+        };
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://newsapi.org")
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+
+        NewsApi newsApi = retrofit.create(NewsApi.class);
+        Call<NewsItemList> call = newsApi.getNews("ru", KEY);
+
+        call.enqueue(callbackNewsItemList);
+
+    }
+
+    void initRecView(){
```

Comment:
We should initialize RecyclerView regardless of the network call. We can make a `setNewsFeed()` method for our adapter and remove list from its constructor parameter. It'll allow us to initialize RecyclerView in `onCreate()` regardless of the network call and use setter in case of the successful network call.